modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -345,25 +347,36 @@ private DepotEntry _addOrUpdateDepotEntry(
 				externalReferenceCode, serviceContext.getCompanyId());
 		}
 
+		DepotEntry depotEntry = null;
+
```

Comment:
Why did you move the variable up? Now it looks like the block under the `group != null` condition will update the variable for later use, but in reality these are two exclusive paths. I think having two separate variables makes it clear that those two code paths are independent.